gke.tf

```diff
@@ -81,7 +81,8 @@ module "gke_auth" {
 module "gke" {
   depends_on                 = [google_compute_instance.gitlab]
   source                     = "terraform-google-modules/kubernetes-engine/google"
-  version                    = "17.3.0"
+  version                    = "22.0.0" # https://github.com/terraform-google-modules/terraform-google-kubernetes-engine
```

Comment:
"The Kubernetes version for the nodes in this pool. Should only be set if auto_upgrade is false"
From readme, you don't specify auto_upgrade.

Why we enforce versions?
Let's select a channel (Rapid) and let the cluster auto update

Reply:
I don't want it to auto upgrade as things might get broken after the upgrade